caffe2/operators/tensorrt_op_trt.cc

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "caffe2/operators/tensorrt_op_trt.h"
+#include "caffe2/core/logging.h"
+
+#include <unordered_map>
+
+namespace caffe2 {
+
+namespace {
+// Note that input of trt tensor is in CHW format, while our tensor is NCHW
+bool CheckDims(
+    const nvinfer1::Dims& nv_dims,
+    const std::vector<TIndex>& c2_dims) {
+  std::cout << "trt_dims: ";
+  for (int i = 0; i < nv_dims.nbDims; ++i) {
+    std::cout << nv_dims.d[i] << ",";
+  }
+  std::cout << std::endl;
+  std::cout << "c2_dims: ";
+  for (int i = 0; i < c2_dims.size(); ++i) {
+    std::cout << c2_dims[i] << ",";
+  }
+  std::cout << std::endl;
+
+  if (nv_dims.nbDims + 1 != c2_dims.size()) {
```

Comment:
just make it all CAFFE_ENFORCE with appropriate error messages

writing to raw cout is bad

Reply:
I forgot to remove the debug string... Will remove them.